File: queuesStacks/stack.mjs
export class Node {
    constructor(data) {
        this.data = data;
        this.next = undefined;
    }
}

export class Stack {
    constructor() {

    }

    pop () {
        const value = this.last;
        if(this.last) {
            this.last = this.last.next;
        }
        return value;
    }
    push (item) {
        if (this.last) {
            item.next = this.last;
            this.last = item;
        } else {
            this.last = item;
        }
    }
    peek () {
        return this.last?.data;
    }
    isEmpty () {
        return !this.last;
    }
}

// const myStack = new Stack();
// console.log('is empty: ' + myStack.isEmpty())
// myStack.push(new Node(1))
// myStack.push(new Node(2))
// myStack.push(new Node(3))
// console.log('is empty: ' + myStack.isEmpty())
// console.log('peek: ' + myStack.peek())
// console.log('pop: ' + myStack.pop()?.data)
// console.log('pop: ' + myStack.pop()?.data)
// console.log('pop: ' + myStack.pop()?.data)

File: queuesStacks/stack.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.last = None
    def pop(self):
        value = self.last
        if self.last is not None:
            self.last = self.last.next
        return value
    def push(self, item):
        if self.last is not None:
            item.next = self.last
            self.last = item
        else:
            self.last = item
    def peek(self):
        return self.last
    def isEmpty(self):
        return self.last is None


# myStack = Stack()
# myStack.push( Node(1))
# myStack.push(Node(2))
# myStack.push(Node(3))
# print('peek ' + str(myStack.peek()))
# print('pop ' + str(myStack.pop().data))
# print('pop ' + str(myStack.pop().data))
# print('pop ' + str(myStack.pop().data))
# print('is empty ' + str(myStack.isEmpty()))
# print('pop ' + str(myStack.pop()))

